cylc/flow/config.py

```diff
@@ -1521,55 +1523,37 @@ def run_mode(self, *reqmodes):
         else:
             return mode
 
-    def check_tasks(self):
-        """Call after all tasks are defined.
-
-        ONLY IF VALIDATING THE SUITE
-        because checking conditional triggers below may be slow for
-        huge suites (several thousand tasks).
```

Comment:
I ran this against a diamond configuration with 100'000 tasks, this method took 0.5% of the wall clock time so I think we're good here.